middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -935,7 +1237,15 @@ let close_functions acc external_env function_declarations =
   let acc = Acc.with_free_names Name_occurrences.empty acc in
   (* CR lmaurer: funs has arbitrary order (ultimately coming from
      function_declarations) *)
-  let funs = Closure_id.Lmap.of_list (Closure_id.Map.bindings funs) in
+  let funs, approximations =
+    let funs, approxs =
+      Closure_id.Map.fold
+        (fun cid (code_id, desc) (funs, approxs) ->
+          (cid, code_id) :: funs, (code_id, desc) :: approxs)
```

Comment:
The `approxs` part of the accumulator seems to be exactly the original `approximations`...

Reply:
It is, but I need a list later and not a `Closure_id.Map`.

Reply:
Are you sure you couldn't just the later code to use maps ? This way you don't have to care about the order of the bindings either.